Windows10 guide for Mattermost-Mobile version. Update _index.md (#667)
mattermost/mattermost-developer-documentation: site/content/contribute/mobile/developer-setup/_index.md

Install the following prerequisite software to develop and build the iOS or Android apps. For macOS, we recommend using [Homebrew](https://brew.sh/) as a package manager.

> #### Install Cygwin (Only for Windows!!!)

Reviewer: ```suggestion
#### Install Cygwin (Windows only)
```

Reviewer: Are there options to install Cygwin for other versions of Windows? If not, one option is to rename this header "Install Cygwin for Windows 10" and then remove the "Windows 10" subheading.

LMK if that makes sense. :)

Author: Hi, thank you for your kind suggestions :D I've searched and found this answer "Windows Vista, Windows Server 2008 and all later versions of Windows up to Windows 10 and Windows Server 2016." from https://cygwin.com/faq/faq.html#faq.what.supported

Reviewer: Brilliant, thank you! What are your thoughts on adding that under the "Install Cygwin" heading? Then the Windows 10 heading will be more targeted.

Author: @justinegeffen Thanks for your suggestion. I will try to put it under the heading.

> ##### Windows10

Reviewer: ```suggestion
##### Windows 10
```

- Install Cygwin from [here](https://www.cygwin.com/)

> - Make sure to install 'make' and 'patch' while installing the Cygwin

Reviewer: ```suggestion
- Make sure to install ``make`` and ``patch`` while installing Cygwin
```

> #### Install [NodeJS](https://nodejs.org/en/).

Reviewer: ```suggestion
#### Install [NodeJS](https://nodejs.org/en/)
```

> This includes NPM which is also needed. Currently version 10.11.0 is recommended with npm 6.4.1. 11.x is **not** working.

Reviewer: ```suggestion
This includes NPM which is also needed. Currently version 10.11.0 is recommended with NPM 6.4.1. 11.x is **not** working.
```

> Install the following prerequisite software to develop and build the iOS or Android apps. For macOS, we recommend using <a href="https://brew.sh/" target="_blank">Homebrew</a> as a package manager.

Reviewer: ```suggestion
Install the following prerequisite software to develop and build the iOS or Android apps. For macOS, we recommend using [Homebrew](https://brew.sh) as a package manager.
```

#### Install Cygwin (Windows only)
##### Windows 10
- [Which versions of Windows does Cygwin support](https://cygwin.com/faq/faq.html#faq.what.supported)?

> - Install Cygwin from [here](https://www.cygwin.com/)

Reviewer: ```suggestion
- Install Cygwin from [here](https://www.cygwin.com/).
```

> - Make sure to install ``make`` and ``patch`` while installing Cygwin

Reviewer: ```suggestion
- Make sure to install `make` and `patch` while installing Cygwin.
```

Install the following prerequisite software to develop and build the iOS or Android apps. For macOS, we recommend using [Homebrew](https://brew.sh/) as a package manager.

#### Install Cygwin (Windows only)
##### Windows 10
- [Which versions of Windows does Cygwin support](https://cygwin.com/faq/faq.html#faq.what.supported)?
- Install Cygwin from [here](https://www.cygwin.com/)
- Make sure to install ``make`` and ``patch`` while installing Cygwin

#### Install [NodeJS](https://nodejs.org/en/)
This includes NPM which is also needed. Currently version 10.11.0 is recommended with NPM 6.4.1. 11.x is **not** working.

##### MacOS

> - To install using Homebrew open a terminal and execute ..

Reviewer: ```suggestion
- To install using Homebrew open a terminal and execute:
```

    different node versions which might be lower than 10.11.0 and may cause problems)
-   Install using NVM by following the instructions [here](https://github.com/creationix/nvm#install-script)
-   Download and install the package from the [NodeJS website](https://nodejs.org/en/)

> ##### Windows10

Reviewer: ```suggestion
##### Windows 10
```

-   Download and install the package from the [NodeJS website](https://nodejs.org/en/)

> #### Install [Watchman](https://facebook.github.io/watchman/). (minimum required version is 4.9.0)

Reviewer: ```suggestion
#### Install [Watchman](https://facebook.github.io/watchman/)

The minimum required version is 4.9.0.
```

Reviewer: Is https://facebook.github.io/watchman/ the correct URL? The URL is mentioned again but in a different order further down the document.

Author: @justinegeffen yes that is the main domain address. Mac can use Homebrew to install Watchman. Linux must install it from source. And I added the instruction for Windows 10 which uses beta package.

    different node versions which might be lower than 10.11.0 and may cause problems)
-   Install using NVM by following the instructions [here](https://github.com/creationix/nvm#install-script)
-   Download and install the package from the [NodeJS website](https://nodejs.org/en/)
##### Windows 10
-   Download and install the package from the [NodeJS website](https://nodejs.org/en/)
#### Install [Watchman](https://facebook.github.io/watchman/)

The minimum required version is 4.9.0.
##### MacOS
- To install using Homebrew open a terminal and execute ..
    ```sh
    $ brew install watchman
    ```
##### Linux
- On Linux you have to build Watchman yourself. See the official [Watchman guide](https://facebook.github.io/watchman/docs/install.html#installing-from-source).
    - Note that you need to increase your ``inotify`` limits for Watchman to work properly
    - If you encounter a warning about a missing C++ compiler you need to install the C++
    extension from your distro's package manager (Ubuntu: g++, RHEL/Fedora: gcc-g++)
##### Windows 10

> - Download the latest package from [here](https://github.com/facebook/watchman/releases/tag/v2020.07.27.00). Note that it's currently beta.

Reviewer: ```suggestion
- Download the latest package from [here](https://github.com/facebook/watchman/releases/tag/v2020.07.27.00). Note that it's currently Beta.
```

-   Using NVM by following the instructions [here](https://github.com/creationix/nvm#install-script).
-   Download and install the package from the [NodeJS website](https://nodejs.org/en).

##### Windows 10
-   Download and install the package from the [NodeJS website](https://nodejs.org/en/)

> #### Install [Watchman](https://facebook.github.io/watchman

Reviewer: ```suggestion
#### Install [Watchman](https://facebook.github.io/watchman0
```

-   Using NVM by following the instructions [here](https://github.com/creationix/nvm#install-script).
-   Download and install the package from the [NodeJS website](https://nodejs.org/en).

##### Windows 10

> -   Download and install the package from the [NodeJS website](https://nodejs.org/en/)

Reviewer 2: @enahum line 46 refers to Linux specifically
![image](https://user-images.githubusercontent.com/13119842/96712259-d7a14c00-136c-11eb-9fc0-aa8f9c86aca0.png)

Perhaps for a later iteration we can consider restructuring this page similar to server developer set up, so that it uses separate tabs for the different OS's
![image](https://user-images.githubusercontent.com/13119842/96712287-e5ef6800-136c-11eb-9010-7b973739d7fe.png)

>     - Note that you need to increase your inotify limits for watchman to work properly

Reviewer: ```suggestion
    - Note that you need to increase your ``inotify`` limits for Watchman to work properly
```

>     - If you encounter a warning about a missing C++ compiler you need to install the c++

Reviewer: ```suggestion
    - If you encounter a warning about a missing C++ compiler you need to install the C++
```

>     extension from you distros package manager (Ubuntu: g++, RHEL/Fedora: gcc-g++)

Reviewer: ```suggestion
    extension from your distro's package manager (Ubuntu: g++, RHEL/Fedora: gcc-g++)
```

> ##### Windows10

Reviewer: ```suggestion
##### Windows 10
```

- Download the latest package from [here](https://github.com/facebook/watchman/releases/tag/v2020.07.27.00). Note that it's currently beta.
#### Install ```react-native-cli``` tools

> Use *npm* to install [React Native CLI Tools](http://facebook.github.io/react-native/docs/understanding-cli.html) globally (minimum required version is 2.0.1)

Reviewer: ```suggestion
Use ``npm`` to install [React Native CLI Tools](https://github.com/react-native-community/cli#react-native-cli) globally (minimum required version is 2.0.1).
```

```sh
$ npm -g install react-native-cli
```
#### Install Ruby

> ##### Windows10

Reviewer: ```suggestion
##### Windows 10
```

- Install Ruby from [here](https://rubyinstaller.org/)

> #### Install git

Reviewer: ```suggestion
#### Install Git
```

> ##### Windows10

Reviewer: ```suggestion
##### Windows 10
```

- Install git from [here](https://git-scm.com/download/win)
#### Install ```bundler --version 2.0.2``` gem
```sh
$ sudo gem install bundler --version 2.0.2
```
#### Obtaining the source code

> We use GitHub to host the source code so we recommend that you install [Git](https://git-scm.com/) to get the source code. Optionally, you can also contribute by submitting [pull requests](https://help.github.com/articles/creating-a-pull-request/). If you do not have git installed you can do so with Homebrew by opening a terminal and executing:

Reviewer: ```suggestion
We use GitHub to host the source code so we recommend that you install [Git](https://git-scm.com/) to get the source code. Optionally, you can also contribute by submitting [pull requests](https://help.github.com/articles/creating-a-pull-request/). If you do not have Git installed you can do so with Homebrew by opening a terminal and executing:
```

- On Linux you have to build Watchman yourself. See the official [Watchman guide](https://facebook.github.io/watchman/docs/install.html#installing-from-source).
   - Note that you need to increase your `inotify` limits for Watchman to work properly.
   - If you encounter a warning about a missing C++ compiler you need to install the C++ extension from your distro's package manager (Ubuntu: g++, RHEL/Fedora: gcc-g++).

##### Windows 10

> - Download the latest package from [here](https://github.com/facebook/watchman/releases/tag/v2020.07.27.00). Note that it's currently in Beta.

Reviewer: I'd leave it as the original purely because there's no guarantee that someone will update this doc if the release status changes. AFAIK we don't have dates attached to our docs (other than commit dates) so "at the time of writing" won't be easy to immediately assess.

> - Download the latest package from [here](https://github.com/facebook/watchman/releases/tag/v2020.07.27.00). Note that it's currently in Beta.

Reviewer 2: Is it important to say that it's in Beta?

#### Install `react-native-cli` tools

> Use `npm` to install [React Native CLI Tools](https://github.com/react-native-community/cli#react-native-cli) globally (minimum required version is 2.0.1).

Reviewer 3: not needed, please remove.

Reviewer 2: Removed

   - Note that you need to increase your `inotify` limits for Watchman to work properly.

>    - If you encounter a warning about a missing C++ compiler you need to install the c++ extension from your distro's package manager (Ubuntu: g++, RHEL/Fedora: gcc-g++).

Reviewer: ```suggestion
   - If you encounter a warning about a missing C++ compiler you need to install the C++ extension from your distro's package manager (Ubuntu: g++, RHEL/Fedora: gcc-g++).
```

##### Windows 10

> - Download the latest package from [here](https://github.com/facebook/watchman/releases/tag/v2020.07.27.00). Note that it's currently Beta.

Reviewer: ```suggestion
- Download the latest package from [here](https://github.com/facebook/watchman/releases/tag/v2020.07.27.00). Note that it's currently in Beta.
```

> #### Install ```react-native-cli``` tools

Reviewer: ```suggestion
#### Install `react-native-cli` tools
```

> Use ``npm`` to install [React Native CLI Tools](https://github.com/react-native-community/cli#react-native-cli) globally (minimum required version is 2.0.1).

Reviewer: ```suggestion
Use `npm` to install [React Native CLI Tools](https://github.com/react-native-community/cli#react-native-cli) globally (minimum required version is 2.0.1).
```

```sh
$ npm -g install react-native-cli
```
#### Install Ruby
##### Windows 10
- Install Ruby from [here](https://rubyinstaller.org/)
#### Install Git
##### Windows 10
- Install git from [here](https://git-scm.com/download/win)

> #### Install ```bundler --version 2.0.2``` gem

Reviewer: ```suggestion
#### Install `bundler --version 2.0.2` gem
```

    ```

4.  Run `make pre-run` in order to install all the dependencies.

>     For Windows10 run `npm install` first then `make pre-run`

Reviewer: ```suggestion
    For Windows 10 run ``npm install`` first then ``make pre-run``
```
